app/controllers/publish_controller.rb

```diff
@@ -20,8 +20,58 @@ def create
     end
   end
 
+  def submitting_pages_not_present?
+    (checkanswers_present? == false) && (confirmation_present? == false)
+  end
+  helper_method :submitting_pages_not_present?
+
+  def cya_page_not_present?
+    (checkanswers_present? == false) && (confirmation_present? == true)
+  end
+  helper_method :cya_page_not_present?
+
+  def confirmation_page_not_present?
+    (checkanswers_present? == true) && (confirmation_present? == false)
+  end
+  helper_method :confirmation_page_not_present?
+
+  def checkanswers_present?
+    form_uuids.include?(find_page_uuid('page.checkanswers'))
+  end
+
+  def confirmation_present?
+    form_uuids.include?(find_page_uuid('page.confirmation'))
+  end
```

Comment:
we can probably have memoized methods for `cya_uuid` and `confirmation_uuid` in the controller. the `service` is  available to do `service.pages.find { |page| page.type == 'page.checkanswers'  }&.uuid` or something similar